### PR TITLE
feat: Added CA-Certificates

### LINKED ISSUE #133: FR: Rename feature 'certificates'
Since the feature _certificates_ provides a means to add _trusted certificate authorities_ to the CloudControl environment, the feature name should also reflect this purpose.

Please change the name og the feature to _ca_certificates_ or _tca_.

Please add your opinions

### PR DESCRIPTION
**Pull request by dploeger**
This is a replacement of certificates, which is deprecated with this.

Includes a feature to set a deprecation in the docs and adds a little not to prefer the FEATURES variable over the USE_ variable.

Fixes #133

### CODE REVIEW

**Review by timdeluxe — CHANGES_REQUESTED** [review]
You meant "little note" (not "little not") didn't you? ;-)

Not a good idea to mix two things in your PR, but since you did i will refer to the second thing you changed: the `USE_` not-recommendations:
First it also should be hinted in the `FEATURES` section: https://github.com/dodevops/cloudcontrol?tab=readme-ov-file#configuring-features. Second, the `<SOMETHING>_VERSION` are still marked as required, but can be given in the `FEATURES` variable as well. Third we should describe why it is recommended to use `FEATURES`: Either we deprecated `USE_` and remove it later, or we leave both options, but then there is no point in recommend `FEATURES` other than it is shorter than `USE_` and we should exactly tell that then.

**Comment by dploeger** [comment]
> Not a good idea to mix two things in your PR, but since you did i will refer to the second thing you changed: the `USE_` not-recommendations: First it also should be hinted in the `FEATURES` section: https://github.com/dodevops/cloudcontrol?tab=readme-ov-file#configuring-features. Second, the `<SOMETHING>_VERSION` are still marked as required, but can be given in the `FEATURES` variable as well. Third we should describe why it is recommended to use `FEATURES`: Either we deprecated `USE_` and remove it later, or we leave both options, but then there is no point in recommend `FEATURES` other than it is shorter than `USE_` and we should exactly tell that then.

I would not deprecate it as the USE_ way can help with automations (like it is used in the test suite currently). Let me rephrase it.

**Comment by dploeger** [comment]
Where do you see that _VERSION is marked as required?

**Comment by timdeluxe** [comment]
> Where do you see that _VERSION is marked as required?

Just search for the string "(required)" in the (generated) docs, you'll find plenty of them. Example:

`Environment HELM_VERSION (required): Valid Helm version to install (e.g. 1.5.4)`

**Comment by dploeger** [comment]
> > Where do you see that _VERSION is marked as required?
> 
> Just search for the string "(required)" in the (generated) docs, you'll find plenty of them. Example:
> 
> `Environment HELM_VERSION (required): Valid Helm version to install (e.g. 1.5.4)`

Ahhh, thaaat's what you mean. Okay. We need to make clear, that a version is required for these features. Either through FEATURES or <feature>_VERSION. I'll take care of it.

**Comment by dploeger** [comment]
I have now made the requirement for a version expressive in the feature descriptor and have adapted the documentation generator. Please take a look.

**Comment by timdeluxe** [comment]
Better but it now misses, that the versions are required. Proposal would be to make it like this:

`VELERO_VERSION (required): Version to install (or use the FEATURES variable instead)`

Or like this:

`VELERO_VERSION: Version to install (or use the FEATURES variable instead) - required`

I prefer the first one, even it is slightly confusing that the <something>_VERSION variable is displayed as required, while it can be as well done through the FEATURES variable. However i think when someone first time uses the FEATURES variable, he/she will get the concept and can extrapolate enough...

**Review by timdeluxe — APPROVED** [review]
Thanks 🎁

File: README.md.gotmpl [hunk 1]
```diff
@@ -320,9 +323,9 @@ to set for the specific environment variable. If a configuration option is requi
 environment declaration using `(required)` after its name and check if the variable is set in your installation script
 and break accordingly if not.
 
-If your feature needs a version specification, the recommended way is to use the environment variable
-`[FEATURE NAME]_VERSION`. This variable is also filled if the *CloudControl* user uses the FEATURES-variable approach
-to enable features.
+If your feature needs a version specification, set "requiresVersion" to true in the feature descriptor. This will
+enable the use of an environment variable `[FEATURE NAME]_VERSION`. This variable is also filled if the *CloudControl*
+user uses the FEATURES-variable approach to enable features.
```
Comment (ThomasMinor) [inline README.md.gotmpl:329]:
```suggestion
If you use the FEATURES variable, you can add the require version separated with a colon to desired feature.
```

**Review by ThomasMinor — CHANGES_REQUESTED** [review]

File: README.md.gotmpl (hunk 1 above)
Comment (dploeger) [reply README.md.gotmpl:329]: This section is about developing new features not the use of it. I'd rather keep the text as is.

Comment (ThomasMinor) [reply README.md.gotmpl:329]: In this case I don't understand why the environment variable is described here. This also covers 'usage'.
But ok.

**Review by ThomasMinor — APPROVED** [review]
